test/support/application_fixture_helper.rb

```diff
@@ -0,0 +1,75 @@
+# typed: ignore
+# frozen_string_literal: true
+
+module ApplicationFixtureHelper
+  TEMP_FIXTURE_DIR = ROOT.join("tmp", "fixtures").to_s
+  DEFAULT_TEMPLATE = :minimal
+
+  def setup_application_fixture
+    @old_working_dir = Dir.pwd
+  end
+
+  def teardown_application_fixture
+    Dir.chdir(@old_working_dir)
+    FileUtils.remove_entry(@app_dir, true) if defined? @app_dir
+  end
+
+  def copy_template(template)
+    copy_dir("test/fixtures/#{template}")
+    Dir.chdir(app_dir)
+  end
+
+  def app_dir
+    copy_template(DEFAULT_TEMPLATE) unless defined? @app_dir
```

Comment:
This line creates a temporary directory and copies over the `minimal` template unless `copy_template(x)` has already been called.  This effectively makes calls to `copy_template(:minimal)` optional, and allows ApplicationFixtureHelper's other methods to work without preconditions. Examples:

```ruby
test "the minimal app is valid" do
  validator = Packwerk::Packwerk::ApplicationValidator(
    config_file_path: config.config_path,  # the `config` method calls `app_dir`
    configuration: config
  )
  assert validator.check_all.ok?
end

test "I can just add a new components/timeline/package.yml file and test the new Package" do
  merge_into_yaml_file("components/timeline/package.yml", { "enforce_privacy" => true })
  package_set = PackageSet.load_all_from(app_dir)
  assert package_set.fetch("components/timeline").enforce_privacy
end
```
I'm not completely sold on the benefits, especially since this behaviour is quite magical.  I can leave it as is, or replace this line with 
```ruby
raise "You need to set up an application fixture by calling `copy_template(:the_template)`." unless defined? @app_dir
```
which makes `copy_template` a required setup step.

Thoughts?

Reply:
I would prefer to be explicit here to avoid a mystery guest from occurring. https://thoughtbot.com/blog/mystery-guest

Reply:
Yeah, I agree. I tried the convenience on for size, but the explicit route is better here.

Reply:
New behaviour:
![image](https://user-images.githubusercontent.com/33674553/100118264-ac19f180-2e43-11eb-95a1-9551077af797.png)
![image](https://user-images.githubusercontent.com/33674553/100118373-c81d9300-2e43-11eb-9bf3-c06de0b9e0c8.png)